src/librustc_typeck/check/method/probe.rs

```diff
@@ -594,25 +631,37 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
         Ok(())
     }
 
+    pub fn matches_return_type(&self, method: &ty::ImplOrTraitItem<'tcx>,
+                               expected: ty::Ty<'tcx>) -> bool {
+        match *method {
+            ty::ImplOrTraitItem::MethodTraitItem(ref x) => {
+                self.probe(|_| {
+                    let output = self.replace_late_bound_regions_with_fresh_var(
```

Comment:
Nit: can you write this as `let (output, _) = ...` instead of doing `output.0` below? I spent a while trying to remember what this `0` field was, would have been more clear that way I think.